games-strategy/julius/julius-1.3.1.ebuild

```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils
+
+DESCRIPTION="An open source re-implementation of Caesar III"
+HOMEPAGE="https://github.com/bvschaik/julius"
+SRC_URI="https://github.com/bvschaik/julius/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="AGPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND="
+	media-libs/libpng
+	media-libs/libsdl2
```

Comment:
Are you sure you don't need [video] and perhaps some other flags set for libsdl?

Reply:
Not sure, I can't test because I can't emerge `media-libs/libsdl2[-video]`:

```
!!! The ebuild selected to satisfy "media-libs/libsdl2" has unmet requirements.
- media-libs/libsdl2-2.0.12-r2::gentoo USE="X alsa dbus gles2 joystick opengl sound threads udev wayland (-aqua) (-custom-cflags) -fcitx4 -haptic -ibus -jack -kms -libsamplerate -nas -oss -pulseaudio -static-libs -tslib -video -vulkan -xinerama -xscreensaver" ABI_X86="(64) -32 (-x32)" CPU_FLAGS_X86="mmx sse sse2 -3dnow" VIDEO_CARDS="(-vc4)"

  The following REQUIRED_USE flag constraints are unsatisfied:
    gles2? ( video ) opengl? ( video )

  The above constraints are a subset of the following complete expression:
    alsa? ( sound ) fcitx4? ( dbus ) gles2? ( video ) ibus? ( dbus ) jack? ( sound ) nas? ( sound ) opengl? ( video ) pulseaudio? ( sound ) vulkan? ( video ) wayland? ( gles2 ) xinerama? ( X ) xscreensaver? ( X )
```

Reply:
Check how `SDL_Init()` is called.

Reply:
Not sure how the lib is used, but it looks like it doesn't say anything about `video`:

```
src/platform/julius.c
360:    if (SDL_Init(SDL_flags) != 0) {
361-        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Could not initialize SDL: %s", SDL_GetError());
362-        return 0;
363-    }
```

Reply:
https://github.com/bvschaik/julius/blob/4ef4dc21b3e1b6a78a51eb9499da0b9f94135364/src/platform/julius.c#L351-L358

You need audio, video, and joystick

Reply:
@DarthGandalf hmmmm, ok, but a broader search does return smth with video:

```
src/platform/screen.c:121:        fullscreen ? "fullscreen" : "windowed", SDL_GetCurrentVideoDriver());
src/platform/screen.c:151:    if (fullscreen && SDL_GetNumVideoDisplays() > 1) {
src/platform/screen.c:263:    if (SDL_GetNumVideoDisplays() > 1) {
src/platform/julius.c:354:    SDL_flags |= SDL_INIT_VIDEO;
```

Reply:
ok, thanks, hadn't seen your reply.

Reply:
@DarthGandalf there's no `audio` USE flag. Would that be alsa OR pulseaudio? Any idea how I can specify OR/XOR USE logic?

Reply:
I think that must be the `sound` one. `alsa` vs `pulseaudio` is what backend will be used, not what SDL subsystems are available.